hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -319,12 +321,18 @@ protected boolean shouldRecreateAndSyncTable() {
 
   private void recreateAndSyncHiveTable(String tableName, boolean useRealtimeInputFormat, boolean readAsOptimized) {
     LOG.info("recreating and syncing the table {}", tableName);
+    Timer.Context timerContext = metrics.getRecreateAndSyncTimer();
     MessageType schema = syncClient.getStorageSchema(!config.getBoolean(HIVE_SYNC_OMIT_METADATA_FIELDS));
     try {
       createOrReplaceTable(tableName, useRealtimeInputFormat, readAsOptimized, schema);
       syncAllPartitions(tableName);
       syncClient.updateLastCommitTimeSynced(tableName);
+      if (Objects.nonNull(timerContext)) {
+        long durationInMs = metrics.getDurationInMs(timerContext.stop());
```

Comment:
Can the conversion happens within `updateRecreateAndSyncMetrics` by taking `timerContext` as the argument?

Reply:
I think it's better to call timerContext.stop here itself. Instead of giving this responsibility to HoodieMetaSyncMetrics class. I can do the conversion to Ms inside HoodieMetaSyncMetrics class.

Reply:
Makes sense.